Salary App/Base/Views/SalaryForm/PensionView.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  PensionComponent.swift
+//  Salary App
+//
+//  Created by Tobi Adegoroye on 27/07/2023.
+//
+
+import UIKit
+
+ 
+class PensionView: UIView {
+    
+    weak var delegate: FormInputDelegate!
+    private var id: String
+
+    private lazy var pensionStackView: UIStackView = {
+        let stackView = UIStackView()
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .equalSpacing
+        stackView.isLayoutMarginsRelativeArrangement = true
+        stackView.layer.cornerRadius = 16
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.spacing = 16
+        return stackView
+    }()
+    
+    private lazy var pensionLbl: UILabel = {
+        let SalaryDescdLbl = UILabel()
+        SalaryDescdLbl.textColor = .black
+        SalaryDescdLbl.text = "Enter pension contribution"
+        SalaryDescdLbl.translatesAutoresizingMaskIntoConstraints = false
+        return SalaryDescdLbl
+    }()
+    
+    private lazy var pensionTextField: UITextField = {
+        let textField = UITextField()
+        textField.layer.borderWidth = 1
+        textField.textColor = .black
+        let paddingView = UIView(frame: CGRect(x: 0, y: 0, width: 10, height: 50))
+        textField.leftView = paddingView
+        textField.placeholder = "Please enter pension Contribution (%)"
+        textField.leftViewMode = .always
+        textField.keyboardType = .asciiCapableNumberPad
+        textField.layer.cornerRadius = 8
+        textField.translatesAutoresizingMaskIntoConstraints = false
+        return textField
+    }()
+    
+    init(id: String) {
+        self.id = id
+        super.init(frame: .zero)
+        setup()
+    }
+    
+   @objc func enterPension() {
+       if let pensionInput = pensionTextField.text,
+          let pensionDelegate = delegate {
```

Comment:
No need to make this part of the optional wrapping, you can use an optional property like

```
if let pensionInput = pensionTextField.text {
  delegate?.didEnter(id: id, input: pensionInput)
}

```